packages/vue-apollo-composable/src/useQuery.ts

```diff
@@ -256,15 +258,10 @@ export function useQueryImpl<
 
     processNextResult(queryResult)
 
-    // Result errors
-    // This is set when `errorPolicy` is `all`
-    if (queryResult.errors?.length) {
-      const e = new Error(`GraphQL error: ${queryResult.errors.map(e => e.message).join(' | ')}`)
-      Object.assign(e, {
-        graphQLErrors: queryResult.errors,
-        networkError: null,
-      })
-      processError(e)
+    // ApolloQueryResult.error may be set at the same time as we get a result
+    // when `errorPolicy` is `all`
+    if (queryResult.error !== undefined) {
```

Comment:
@backbone87 @Akryum 

Is this change correct? I've tried the latest stable @apollo/client@3.3 and the next version 3.4.1.

Both of them don't have have `.error` set, they only have `.errors` set.

This means error is never going to trigger `onError`.

Here is my query options

```
    defaultOptions: {
      watchQuery: {
        fetchPolicy: 'no-cache',
        errorPolicy: 'all',
      },
      query: {
        fetchPolicy: 'no-cache',
        errorPolicy: 'all',
      },
      mutate: {
        errorPolicy: 'all',
      },
    },
```

Reply:
I have tested with @apollo/client@3.4.1

`.error` is set when `errorPolicy=none` and it is not set when `errorPolicy=all`

Reply:
I can see the relevant PRs to the error field
https://github.com/apollographql/apollo-client/commit/513d733cf0162c86d2b7d56547c20bb03287a472#diff-80ebcd3dc48e85c12b2c8484c5355662ab98ceea358660eb3af536c584f6fa44R22

Reply:
my assumption was that error contains an ApolloError instance that already wraps ApolloQueryResult.errors, seems like this is not the case... so we need to create the ApolloError for ApolloQueryResult.errors outselves.

Reply:
@backbone87 I have the similar assumption but I can't find anything in official clearly says should `onError` be triggered or not when `errorPolicy=all`. Looks like this might be an unclear behaviour on `apollo/client` itself.

I can see different PRs, some assume `onError` should be triggered and some assume it should not.

Reply:
in the end vue-apollo wraps all this stuff anyway, so it can define its behavior (yay more convolution :))